src/js/common/schemaform/ApplicationStatus.jsx

```diff
@@ -0,0 +1,139 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import moment from 'moment';
+import { connect } from 'react-redux';
+
+import { formLinks, formTitles } from '../../user-profile/helpers';
+import LoadingIndicator from '../../common/components/LoadingIndicator';
+import ProgressButton from '../../common/components/form-elements/ProgressButton';
+import Modal from '../../common/components/Modal';
+import { removeFormApi } from '../../common/schemaform/sip-api';
+
+export class ApplicationStatus extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      modalOpen: false,
+      loading: false
+    };
+
+    moment.updateLocale('en', {
+      meridiem: (hour) => {
+        if (hour < 12) {
+          return 'a.m.';
+        }
+        return 'p.m.';
+      }
+    });
+  }
+
+  removeForm = () => {
+    this.setState({ modalOpen: false, loading: true });
+    removeFormApi(this.props.formId)
+      // Swallow any errors and redirect anyway
+      .catch(x => x)
+      .then(() => {
+        window.location.href = formLinks[this.props.formId];
+      });
+  }
+
+  toggleModal = () => {
+    this.setState({ modalOpen: !this.state.modalOpen });
+  }
+
+  render() {
+    const { formId, profile, login, applyText, showApplyButton } = this.props;
+
+    if (profile.loading || this.state.loading) {
+      const message = profile.loading
+        ? 'Checking your application status.'
+        : 'Deleting your form.';
+
+      return (
+        <div className="sip-application-status">
+          <LoadingIndicator message={message}/>
+        </div>
+      );
+    }
+
+    const savedForm = profile.savedForms.find(({ form }) => form === formId);
+
+    if (login.currentlyLoggedIn && savedForm) {
+      const { last_updated: lastSaved, expires_at: expirationTime } = savedForm.metadata;
+      const expirationDate = moment.unix(expirationTime).format('M/D/YYYY');
+      const isExpired = moment(expirationDate).isBefore();
+
+      const [firstLetter, ...restOfTitle] = formTitles[formId];
+      const cardTitle = `${firstLetter.toUpperCase()}${restOfTitle.join('')} application in progress`;
```

Comment:
Should we ensure `restOfTitle` is all lower case, or will it be okay if we have some caps (like we often do for 'Veteran')?

Reply:
I think it's ok if it has some caps, I just wanted to make sure the first letter was capitalized.